pkg/agent/secondarynetwork/podwatch/k8s/podwatch_controller.go

```diff
@@ -0,0 +1,338 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podwatch
+
+import (
+	"context"
+	"encoding/json"
+
+	"errors"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	invoke "github.com/containernetworking/cni/pkg/invoke"
+	current "github.com/containernetworking/cni/pkg/types/current"
+	netdefclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
+	netdefutils "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/utils"
+
+	cnipodcache "antrea.io/antrea/pkg/agent/secondarynetwork/cnipodcache"
+	ipam "antrea.io/antrea/pkg/agent/secondarynetwork/ipam"
+	cniserver "antrea.io/antrea/pkg/agent/cniserver"
+)
+
+const (
+	controllerName = "AntreaAgentPodWatchController"
+	minRetryDelay  = 2 * time.Second
+	maxRetryDelay  = 120 * time.Second
+	numWorkers     = 4
+)
+
+const (
+	networkAttachDefAnnotationKey = "k8s.v1.cni.cncf.io/networks"
+	sriov                         = "sriov"
+	netDev                        = "eth1"
+)
+
+type PodWatchController struct {
```

Comment:
I remember Antonin suggested to rename it to PodController.

And in my mind, we should put this file under /podwatch directly (remove /k8s), and rename it to "controller.go".